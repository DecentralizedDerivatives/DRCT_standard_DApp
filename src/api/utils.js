import axios from 'axios';

const extractData = response => {
  const milliseconds = 1000;
  return response.data.Data.map(obj => [obj.time * milliseconds, obj.close]);
};

export const get = async url => {
  let response, error;

  try {
    response = await axios.get(url);
  } catch (err) {
    error = err;
  }

  return response ? extractData(response) : error;
};
