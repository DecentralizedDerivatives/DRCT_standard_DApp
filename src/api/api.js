import {get} from './utils';

const btc = {
  get: () =>
    get(
      'https://min-api.cryptocompare.com/data/histohour?fsym=BTC&tsym=USD&limit=2000'
    ),
};

const eth = {
  get: () =>
    get(
      'https://min-api.cryptocompare.com/data/histohour?fsym=ETH&tsym=USD&limit=2000'
    ),
};

export default {
  eth,
  btc,
};
