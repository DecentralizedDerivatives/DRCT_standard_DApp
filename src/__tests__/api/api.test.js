import mockAxios from 'jest-mock-axios';

import api from '../../api/api';

afterEach(() => {
  // cleaning up the mess left behind the previous test
  mockAxios.reset();
});

it('should call BTC url', () => {
  const priceData = api['btc'].get();
  expect(mockAxios.get.mock.calls).toMatchSnapshot();
});

it('should call ETH url', () => {
  const priceData = api['eth'].get();
  expect(mockAxios.get.mock.calls).toMatchSnapshot();
});

it('should call minute BTC url', () => {
  const priceData = api['btcMinute'].get();
  expect(mockAxios.get.mock.calls).toMatchSnapshot();
});

it('should call minute ETH url', () => {
  const priceData = api['ethMinute'].get();
  expect(mockAxios.get.mock.calls).toMatchSnapshot();
});
