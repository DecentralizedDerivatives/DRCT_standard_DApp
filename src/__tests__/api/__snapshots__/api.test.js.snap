// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call BTC url 1`] = `
Array [
  Array [
    "https://min-api.cryptocompare.com/data/histohour?fsym=BTC&tsym=USD&limit=2000",
  ],
]
`;

exports[`should call ETH url 1`] = `
Array [
  Array [
    "https://min-api.cryptocompare.com/data/histohour?fsym=ETH&tsym=USD&limit=2000",
  ],
]
`;

exports[`should call minute BTC url 1`] = `
Array [
  Array [
    "https://min-api.cryptocompare.com/data/histominute?fsym=BTC&tsym=USD&limit=2000",
  ],
]
`;

exports[`should call minute ETH url 1`] = `
Array [
  Array [
    "https://min-api.cryptocompare.com/data/histominute?fsym=ETH&tsym=USD&limit=2000",
  ],
]
`;
